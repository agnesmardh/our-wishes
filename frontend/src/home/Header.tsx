import React from 'react';
import { Button, Form, Nav, Navbar } from 'react-bootstrap';
import { useAuth } from '../common/auth/ProvideAuth';
import styled from 'styled-components';
import { SmallLogo } from './SmallLogo';
import { PersonCircle } from 'react-bootstrap-icons';

export const Header: React.FC = () => {
  const { signOut } = useAuth();
  return (
    <Navbar bg="light" expand="lg">
      <Navbar.Brand href="/">
        <SmallLogo /> <HeaderBrandTitle>Our Wishes</HeaderBrandTitle>
      </Navbar.Brand>
      <Nav className="mr-auto" />
      <UserInfoContainer>
        <UserAvatar size={32} />
        <span>Username</span>
      </UserInfoContainer>
      <Form inline>
        <Button variant="outline-success" onClick={async () => await signOut()}>
          Sign out
        </Button>
      </Form>
    </Navbar>
  );
};

const HeaderBrandTitle = styled.span`
  font-family: PermanentMarker, serif;
  display: inline;
`;

const UserInfoContainer = styled.div`
  margin-right: 10px;
`;

const UserAvatar = styled(PersonCircle)`
  margin-right: 10px;
`;
