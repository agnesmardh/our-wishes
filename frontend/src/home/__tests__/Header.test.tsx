import React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../Header';
import userEvent from '@testing-library/user-event';

const mockSignOutFn = jest.fn();

jest.mock('../../common/auth/ProvideAuth', () => {
  return {
    useAuth: () => ({
      signOut: mockSignOutFn
    })
  };
});

describe('<Header/>', () => {
  const renderHeader = () => {
    return render(<Header />);
  };

  it('should render <Header/> and match the snapshot', () => {
    const header = renderHeader();
    expect(header).toMatchSnapshot();
  });

  it('should call Auth.signOut when signOut button is pressed', () => {
    const header = renderHeader();

    const signOutButton = header.getByRole('button', { name: /sign out/i });
    userEvent.click(signOutButton);
    expect(mockSignOutFn).toHaveBeenCalledTimes(1);
  });
});
