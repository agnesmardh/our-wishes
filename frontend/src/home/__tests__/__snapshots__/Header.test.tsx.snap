// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render <Header/> and match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="logo"
            height="42"
            src="logo-small.svg"
            width="42"
          />
           
          <span
            class="sc-bdnxRM cqFNfF"
          >
            Our Wishes
          </span>
        </a>
        <div
          class="mr-auto navbar-nav"
        />
        <div
          class="sc-gtsrHT kBuNnx"
        >
          <svg
            class="sc-dlnjwi ixexop"
            fill="currentColor"
            height="32"
            viewBox="0 0 16 16"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
            />
            <path
              d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm8-7a7 7 0 0 0-5.468 11.37C3.242 11.226 4.805 10 8 10s4.757 1.225 5.468 2.37A7 7 0 0 0 8 1z"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            Username
          </span>
        </div>
        <form
          class="form-inline"
        >
          <button
            class="btn btn-outline-success"
            type="button"
          >
            Sign out
          </button>
        </form>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt="logo"
          height="42"
          src="logo-small.svg"
          width="42"
        />
         
        <span
          class="sc-bdnxRM cqFNfF"
        >
          Our Wishes
        </span>
      </a>
      <div
        class="mr-auto navbar-nav"
      />
      <div
        class="sc-gtsrHT kBuNnx"
      >
        <svg
          class="sc-dlnjwi ixexop"
          fill="currentColor"
          height="32"
          viewBox="0 0 16 16"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
          />
          <path
            d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm8-7a7 7 0 0 0-5.468 11.37C3.242 11.226 4.805 10 8 10s4.757 1.225 5.468 2.37A7 7 0 0 0 8 1z"
            fill-rule="evenodd"
          />
        </svg>
        <span>
          Username
        </span>
      </div>
      <form
        class="form-inline"
      >
        <button
          class="btn btn-outline-success"
          type="button"
        >
          Sign out
        </button>
      </form>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
